### PR TITLE
Add Out of Stock alert

### LINKED ISSUE #951: Design request: Change OOS Notify Me box color
>Feedback from the team: Design request to change the color of the OOS notice. Blue vs. immediately obvious red.

https://react-commerce.vercel.app/products/macbook-pro-14-2021-a2442-right-fan
image

Previously, before expanding to input email
image

cc: @jeffsnyder

### PR DESCRIPTION
**Pull request by dhmacs**
closes #951 

This PR adds an out of stock alert box as requested in #951 and contains also a couple of color corrections.

## QA

1. Open Vercel preview
2. Verify that the new design for Out of Stock has been implemented
    Screenshot 2023-01-31 at 14 36 05

### CODE REVIEW

**Review by masonmcelvain — DISMISSED** [review]
CR ✌🏻 When this merges I will open a followup pull to make the "Not for Sale" alert use this new component.

**Comment by dhmacs** [comment]
@masonmcelvain It's the standard chakra alert component, I've changed the theme to reflect the mockup colors. Reading the code it seems like you just need to remove the color override props and just let the component pick those from the theme 👍

**Comment by ardelato** [comment]
deploy_block ☁️ We will need to update the the `add_to_cart-Out of Stock` test to reflect this change

**Comment by dhmacs** [comment]
@ardelato are you working on it or can I do it? The test seems to pass in CI though 🤔

**Comment by ardelato** [comment]
> @ardelato are you working on it or can I do it? The test seems to pass in CI though 🤔

If you want to work on it, you can. 

Although the test is passing, the spec for an `out of stock` product has changed. From the changes in this pull, it seems we should now expect a new component with the text `This item is currently Out of Stock` to render.

**Comment by erinemay** [comment]
Is there a reason why previews haven't been deployed?

**Review by masonmcelvain — APPROVED** [review]
CR ✌🏻

**Comment by ardelato** [comment]
> I'll let you modify it as you see fit, I've just imported the latest changed from master.

Roger that, un_deploy_block ⚡

**Comment by erinemay** [comment]
QA 🦖 Looks good. The only difference is the OOS alert above the existing box. Emails are still getting saved, and there's no other change in the ui or ux of signing up for product notifications. 
Looks good on desktop/tablet/mobile iOS and Android. 

deploy_block 🧷 on a couple grammatical changes

`Enter your email address below and we will notify you when this product is back in stock.` should have a comma -> `Enter your email address below, and we will notify you when this product is back in stock.`

`This item is currently Out of Stock` needs a period at the end -> `This item is currently Out of Stock.`

File: frontend/templates/product/sections/ProductSection/AddToCart/index.tsx [hunk 1]
```diff
@@ -113,6 +115,17 @@ export function AddToCart({ product, selectedVariant }: AddToCartProps) {
...
                )}
+               <Alert status="warning" mb="3">
+                  <FaIcon
+                     icon={faExclamationCircle}
+                     h="5"
+                     mr="2"
+                     color="amber.600"
+                  />
+                  <span>
+                     This item is currently <strong>Out of Stock</strong>
```
Comment (erinemay) [inline frontend/templates/product/sections/ProductSection/AddToCart/index.tsx:126]:
```suggestion
                     This item is currently <strong>Out of Stock</strong>.
```

File: frontend/templates/product/sections/ProductSection/AddToCart/NotifyMeForm.tsx [hunk 2]
```diff
@@ -80,8 +80,12 @@ export function NotifyMeForm({ sku }: NotifyMeFormProps) {
          py="4"
       >
          <form onSubmit={handleFormSubmit}>
-            <Text color="brand.500">
-               This item is currently <strong>Out of Stock</strong>
+            <Text fontWeight="semibold" mb="1.5">
+               Notify me when it is back in stock!
+            </Text>
+            <Text>
+               Enter your email address below and we will notify you when this
```
Comment (erinemay) [inline frontend/templates/product/sections/ProductSection/AddToCart/NotifyMeForm.tsx:87]:
```suggestion
               Enter your email address below, and we will notify you when this
```

**Comment by erinemay** [comment]
Oops. I just wanted to fix some grammatical issues and now we're failing tests. Sorry @dhmacs.

**Comment by masonmcelvain** [comment]
@erinemay Do you mind if I just rollback those commits and merge this? Maybe we can make the grammatical corrections in #1325 when we update the test

**Comment by masonmcelvain** [comment]
CR ✌🏻 QA ✌🏻 carry forward, since https://github.com/iFixit/react-commerce/pull/1317/commits/18657a06555310318310630676b910a5979bf015 already had CR and QA

**Comment by masonmcelvain** [comment]
and un_deploy_block ✌🏻

**Comment by masonmcelvain** [comment]
I looked at the failing tests and decided I would try to fix them instead of rolling back the changes, sorry for the confusion @erinemay.

**Review by masonmcelvain — APPROVED** [review]
CR ✌🏻 of Erin's changes

**Comment by erinemay** [comment]
@masonmcelvain Thank you!!! Please always just roll back if I mess something up. I'll leave things as suggestions from now on 😢 

QA 👍

**Comment by masonmcelvain** [comment]
No worries, they were good suggestions and I don't think you should be deterred from pushing commits in the future. Our tests are just very strict apparently.

**Comment by dhmacs** [comment]
> Our tests are just very strict apparently.

I agree, we should try to make them more resilient to small changes 👍

**Pull request merged**